File: python/sort/insertion/insertion.py
def move(arr, i, k):
    num = arr[i]
    while k > -1:
        if num < arr[k]:
            arr[i] = arr[k]
            i -= 1
            k -= 1
    arr[k] = num


def insertion(arr):
    k = 0
    for i in range(1, len(arr), 1):
        if arr[i] < arr[i - 1]:
            move(arr, i, k)
            k += 1

    return arr


print(insertion([8, -1, 4, 5, 1, 9, 7, 3, 0]))

File: python/sort/insertion/insertion.js
function move(arr, i) {
    num = arr[i];
    prev = i - 1

    while (num < arr[prev]) {
        arr[i--] = arr[prev--];
    }

    arr[i] = num
}

function insertion(arr) {
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] < arr[i - 1]) {
            move(arr, i);
        }
    }

    return arr;
}

def insertion(arr):
    for i in range(1, len(arr)):
        if arr[i] < arr[i - 1]:
            move(arr, i)
    return arr


//arr = [100, 0, 5, 0, -9, 4, 1, -100];
arr = [5, 0];
//console.log("start", arr);
console.log(insertion(arr));
